pkg/controller/elasticsearch/observer/manager.go

```diff
@@ -67,13 +67,20 @@ func (m *Manager) getObserver(key types.NamespacedName) (*Observer, bool) {
 
 // Observe gets or create a cluster state observer for the given cluster
 // In case something has changed in the given esClient (eg. different caCert), the observer is recreated accordingly
-func (m *Manager) Observe(ctx context.Context, cluster esv1.Elasticsearch, esClient client.Client, isServiceReady bool) *Observer {
+func (m *Manager) Observe(ctx context.Context, cluster esv1.Elasticsearch, esClientProvider func(client.Client) client.Client, isServiceReady bool) *Observer {
```

Comment:
We already have an EsClientProvider type:

https://github.com/elastic/cloud-on-k8s/blob/b6eca617087885b672391bc889af66ad59154623/pkg/controller/autoscaling/elasticsearch/controller.go#L42

Can we consolidate your changes with this existing?

Reply:
I am not sure it makes sense. I agree that the there is an unfortunate naming overlap. But here we actually have different function signature `func(esclient.Client) esclient.Client`. We can think about a better name maybe?

Reply:
Hmm, I see. Let's keep it that way.